### PR TITLE
Add new PMOTS: Roulettemons

### PR DESCRIPTION
**Pull request by charizard8888**
April 1: https://www.smogon.com/forums/threads/pet-mod-of-the-season-voting-stage.3661834/
Coded by [DuoM2](https://www.smogon.com/forums/members/duom2.364743/) using Scoopapa's [Tool](https://www.smogon.com/forums/threads/pet-mods-general-discussion-thread.3621931/page-10#post-7963043)

Are the dex num fine?

### CODE REVIEW

File: config/formats.js [hunk 1]
```diff
@@ -839,21 +839,19 @@ let Formats = [
 		column: 2,
 	},
 	{
-		name: "[Gen 7 Pet Mod] Clean Slate: Micro",
-		desc: `A brand new "micrometagame" created from scratch, with the ultimate goal of creating a unique, compact metagame different from any other tier.`,
+		name: "[Gen 8] Roulettemons",
+		desc: `A competitive metagame made up of brand new Pokémon that are randomly generated on every competitive aspect (typing, moves, stats, abilities).`,
```
Comment (KrisXV) [inline config/formats.js:843]:
```suggestion
		desc: `A competitive metagame made up of brand new Pok&eacute;mon that have randomly generated moves, stats, abilities, and types.`,
```

File: config/formats.js [hunk 2]
```diff
@@ -839,21 +839,19 @@ let Formats = [
...
 		threads: [
-			`<a href="https://www.smogon.com/forums/threads/3652540/">Clean Slate: Micro</a>`,
+			`<a href=https://www.smogon.com/forums/threads/.3649106/>Roulettemons</a>`,
```
Comment (KrisXV) [inline config/formats.js:845]:
```suggestion
			`<a href="https://www.smogon.com/forums/threads/3649106/">Roulettemons</a>`,
```

File: config/formats.js [hunk 3]
```diff
@@ -839,21 +839,19 @@ let Formats = [
...
 		],
-
-		mod: 'cleanslatemicro',
-		ruleset: ['Standard Pet Mod'],
+		mod: 'roulettemons',
+		ruleset: [ 'Dynamax Clause', 'Obtainable Moves', 'Sleep Clause Mod', 'Species Clause', 'Moody Clause', 'Evasion Moves Clause', 'Endless Battle Clause', 'HP Percentage Mod', 'Cancel Mod', 'Team Preview', 'Swagger Clause', 'Baton Pass Clause'],
+		banlist: ['All Pokemon'],
```
Comment (KrisXV) [inline config/formats.js:850]:
```suggestion
		mod: 'roulettemons',
		ruleset: ['Standard Pet Mod'],
```

The point of Standard Pet Mod is to put all of the custom rules and bans in the ruleset.

File: config/formats.js [hunk 4]
```diff
@@ -839,21 +839,19 @@ let Formats = [
...
 		unbanlist: [
-			'Crobat', 'Dragalge', 'Dugtrio-Alola', 'Farfetch\'d', 'Galvantula', 'Heracross-Base', 'Kyurem-Base', 'Ludicolo',
-			'Magearna-Base', 'Malamar', 'Ninetales-Base', 'Pupitar', 'Purugly', 'Rotom-Base', 'Rotom-Heat', 'Rotom-Mow',
-			'Rotom-Wash', 'Torterra', 'Type: Null', 'Umbreon', 'Wailord',
+			'Koatric', 'Aquazelle', 'Salamalix', 'Brawnkey', 'Stuneleon', 'Chillyte', 'Chillyte-Mega', 'Eartharoo', 'Crazefly', 'Electritar', 'Aquatopus', 'Scorpita', 'Baloon', 'Kinesel', 'Glacida', 'Pidgeotine', 'Gorilax', 'Albatrygon', 'Chillvark', 'Komodith', 'Giranium', 'Flamyle', 'Voltecta', 'Ostria', 'Ninjoth', 'Herbigator', 'Anteros', 'Gladiaster', 'Hyperoach', 'Barracoth', 'Toados', 'Voltarak', 'Mosqung', 'Flamepion', 'Hyenix', 'Rhinolite', 'Bellena', 'Falcola', 'Beanium', 'Lemotic', 'Biceon', 'Skeleray', 'Specyte', 'Ramron', 'Panthee', 'Blastora', 'Balar', 'Dropacle',
 		],
 		onSwitchIn(pokemon) {
-			this.add('-start', pokemon, 'typechange', pokemon.species.types.join('/'), '[silent]');
+			this.add('-start', pokemon, 'typechange', pokemon.template.types.join('/'), '[silent]');
```
Comment (KrisXV) [inline config/formats.js:855]:
```suggestion
			this.add('-start', pokemon, 'typechange', pokemon.species.types.join('/'), '[silent]');
```

File: config/formats.js [hunk 5]
```diff
@@ -839,21 +839,20 @@ let Formats = [
 		column: 2,
 	},
 	{
-		name: "[Gen 7 Pet Mod] Clean Slate: Micro",
-		desc: `A brand new "micrometagame" created from scratch, with the ultimate goal of creating a unique, compact metagame different from any other tier.`,
+		name: "[Gen 8] Roulettemons",
```
Comment (KrisXV) [inline config/formats.js:842]:
```suggestion
		name: "[Gen 8 Pet Mod] Roulettemons",
```

**Comment by charizard8888** [comment]
Oh well looks like we code a bit too much simply

**Comment by charizard8888** [comment]
Big thanks to Kris and sorry for the troubles

**Comment by charizard8888** [comment]
Question: Where to add sprites in the client? /data/sprites/ is hidden for me I think

**Comment by Zarel** [comment]
`/data/sprites/` is not MIT licensed, and so isn't in the repository at all.

**Comment by charizard8888** [comment]
Then is there a folder where sprites have to be added?

**Comment by Zarel** [comment]
You can just send them to me.

**Comment by KrisXV** [comment]
I'm pretty sure the entire formats-data file is unnecessary.

**Comment by scoopapa** [comment]
formats-data crash
Tried removing it on Glitch and got this error

**Comment by charizard8888** [comment]
AquaticPanic has the sprites on his PC, he'll attach them soon.

**Comment by AquaticPanic** [comment]
[sprites.zip](https://github.com/smogon/pokemon-showdown/files/4430623/sprites.zip)

**Comment by charizard8888** [comment]
Hey @TheImmortal we have everything ready apart from 10 sprites which are under progress.

**Comment by charizard8888** [comment]
> You can just send them to me.

Hey @Zarel most of the sprites are here. Also how can we set up the Showdown Client to have the details of Roulettemons? The test client which I used to use earlier had some different files from the Showdown Client.

> [sprites.zip](https://github.com/smogon/pokemon-showdown/files/4430623/sprites.zip)

**Pull request merged**